apps/spawn_sdk/lib/system/spawn_system.ex

```diff
@@ -234,41 +72,135 @@ defmodule SpawnSdk.System.SpawnSystem do
 
     case Actors.spawn_actor(build_spawn_req(system, actor_name, actor_mod), opts) do
       {:ok, %SpawnResponse{status: status}} ->
-        Logger.debug("Actor Spawned succeed.. Status: #{inspect(status)}")
+        Logger.debug("Actor Spawned successfully. Status: #{inspect(status)}")
+
+        merge_cache_actors(system, new_state)
 
         {:ok, new_state}
 
       error ->
-        {:error, "Actors Spawned failed. Error #{inspect(error)}"}
+        {:error, "Actors Spawned failing. Error #{inspect(error)}"}
     end
   end
 
-  defp do_invoke(system, actor, command, payload, options) do
+  @impl SpawnSdk.System
+  @doc "hey"
+  def invoke(actor_name, invoke_opts \\ []) do
+    system = Keyword.get(invoke_opts, :system, nil)
+    command = Keyword.get(invoke_opts, :command, nil)
+    payload = Keyword.get(invoke_opts, :payload, nil)
+    async = Keyword.get(invoke_opts, :async, false)
+    lazy_spawn = Keyword.get(invoke_opts, :lazy_spawn, nil)
+
+    if lazy_spawn do
+      spawn_actor(actor_name, system: system, actor: lazy_spawn)
+    end
+
     opts = [host_interface: SpawnSdk.Interface]
-    async = Keyword.get(options, :async, false)
 
     req =
       InvocationRequest.new(
         system: %Eigr.Functions.Protocol.Actors.ActorSystem{name: system},
-        actor: %Eigr.Functions.Protocol.Actors.Actor{name: actor},
+        actor: %Eigr.Functions.Protocol.Actors.Actor{name: actor_name},
         value: any_pack!(payload),
         command_name: command,
         async: async
       )
 
-    if async do
-      Actors.invoke(req, opts)
-      {:ok, :async}
-    else
-      case Actors.invoke(req, opts) do
-        {:ok, %ActorInvocationResponse{value: value}} ->
-          {:ok, unpack_unknown(value)}
+    case Actors.invoke(req, opts) do
+      {:ok, :async} -> {:ok, :async}
+      {:ok, %ActorInvocationResponse{value: value}} -> {:ok, unpack_unknown(value)}
+      error -> error
+    end
+  end
+
+  def call(invocation, entity_state, default_methods) do
+    %ActorInvocation{
+      actor_name: name,
+      actor_system: system,
+      command_name: command,
+      value: value
+    } = invocation
+
+    %EntityState{
+      actor: %Actor{state: actor_state} = actor
+    } = entity_state
+
+    actor_state = actor_state || %{}
+    current_state = Map.get(actor_state, :state)
+    actor_instance = get_cached_actor(system, name)
```

Comment:
this is a potential problem, if the actor is `abstract` and was not spawned in this same node, it would not call here due to trying to find in `:ets`, the same would happen if it was a local GenServer though.

Not having this would mean that we somehow know in the ActorInvocation{} what module we should be calling, it would mean a change in the proto

Reply:
When there is an invocation the system should use Actors.invoke. In turn, Actors.invoke searches the ActorRegistry for nodes that have that actor registered and invokes the actor on any node that has the actor registered. The sdk should never ask itself whether or not it owns that actor, as it is not the sdk's responsibility but the Actors to know which node to invoke. The sdk when it receives an ActorInvocation can look at itself and see if it has an actor registered or trust the proxy.